File: 프로그래머스/unrated/181928. 이어 붙인 수/이어 붙인 수.java
class Solution {
    public int solution(int[] num_list) {
        StringBuffer even = new StringBuffer();
        StringBuffer odd = new StringBuffer();
        for(int i : num_list) {
            if (i % 2 == 0) {
                even.append(i);
            } else {
                odd.append(i);
            }
        }
        return Integer.parseInt(even.toString()) + Integer.parseInt(odd.toString());
    }
}

File: 프로그래머스/unrated/181928. 이어 붙인 수/이어 붙인 수.js
function solution(num_list) {
    var odd = "";
    var even = "";
    for (num of num_list) {
        if (num % 2 === 0) {
            even += num.toString()
        } else {
            odd  += num.toString()
        }
    }
    return Number(even) + Number(odd);
}